### PR TITLE
Kill whitebox crash test if it is 15 minutes over the limit

### PR DESCRIPTION
**Pull request by siying**
Summary:
Whitebox crash test can run significantly over the time limit for test slowness or no kiling points. This indefinite job can create problem when this test is periodically scheduled as a job. Instead, kill the job if it is 15 minutes over the limit.
Refactor the code slightly to consolidate the code for executing commands for white and black box tests.

Test Plan: Run both of black and white box tests with both of natual and explicit kill condition.

### CODE REVIEW

File: tools/db_crashtest.py [hunk 1]
```diff
@@ -461,6 +461,25 @@ def inject_inconsistencies_to_db_dir(dir_path):
...
+    child = subprocess.Popen(cmd, stderr=subprocess.PIPE,
+                             stdout=subprocess.PIPE)
+    print("Running db_stress with pid=%d: %s\n\n"
+          % (child.pid, ' '.join(cmd)))
+
+    try:
+        outs, errs = child.communicate(timeout=timeout)
+        hit_timeout = False
+        print("WARNING: db_stress ended before kill: exitcode=%d\n"
+              % child.returncode)
```
Comment (jay-zhuang) [inline tools/db_crashtest.py:474]: Is there a way to know if it's still running or hang?
Should we dump stackcalls before killing it?

**Review by jay-zhuang — APPROVED** [review]
LGTM

File: tools/db_crashtest.py (hunk 1 above)
Comment (siying) [reply tools/db_crashtest.py:474]: I think it can be a good idea. If someone can pick it up and finish it, that could be great.

**Review by ammarfaizi2 — APPROVED** [review]

File: tools/db_crashtest.py [hunk 2]
```diff
@@ -474,47 +493,26 @@ def blackbox_crash_main(args, unknown_args):
...
+            print('Exit Before Killing') 
+            print('stdout:')
+            print(outs)
+            print('stderr:')
+            print(errs)
+            sys.exit(2)
+
+        for line in errs.split('\n'):
+            if line != '' and  not line.startswith('WARNING'):
                 run_had_errors = True
```
Comment (thatsafunnyname) [inline tools/db_crashtest.py:512]: I noticed https://lgtm.com/projects/g/facebook/rocksdb/snapshot/b215f1a83226f111ff52305987af93564272b7d3/files/tools/db_crashtest.py?sort=name&dir=ASC&mode=heatmap#xf254f528ad18f108:1 is correctly reporting that `run_had_errors` is now unused.  Xref: https://github.com/facebook/rocksdb/pull/8599

**Pull request closed without merging**